hal/src/rtl872x/ble_hal.cpp

```diff
@@ -2458,7 +2458,7 @@ void BleGap::handleConnectionStateChanged(uint8_t connHandle, T_GAP_CONN_STATE n
             if (connecting_ && connection.info.role == BLE_ROLE_CENTRAL) {
                 connecting_ = false;
                 // See: handleMtuUpdated()
-                // os_semaphore_give(connectSemaphore_, false);
+                os_semaphore_give(connectSemaphore_, false);
```

Comment:
The gist of the BLE changes are that we give the `connectSempahore_` after connecting in the central role, rather than when the MTU is updated? How was this causing problems? We were giving the semaphore on MTU negotiation before connection was complete?

Reply:
This should not be here, covered by a different PR. This was a temporary fix. I'll get rid of it.